src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -2001,9 +2001,17 @@ private void ParseUsingKeyword(SyntaxListBuilder<RazorSyntaxNode> builder, CShar
                 // using Identifier ==> Using Declaration
                 if (!topLevel)
                 {
-                    Context.ErrorSink.OnError(
-                        RazorDiagnosticFactory.CreateParsing_NamespaceImportAndTypeAliasCannotExistWithinCodeBlock(
-                            new SourceSpan(block.Start, block.Name.Length)));
+                    // using Variable Declaration
+
+                    if (!Context.FeatureFlags.AllowUsingVariableDeclarations)
+                    {
+                        Context.ErrorSink.OnError(
```

Comment:
So turns out we already supported using variable declaration syntax. We just tried to be smart and notify users of errors that C# already would have >_<. This PR's contents are really just about not causing an error in using variable declaration syntax any more.

Also added a boat load of C# 8.0 tests.